.github/workflows/build-and-publish.yml

```diff
@@ -113,11 +113,57 @@ jobs:
           name: workspace
           path: ./
   
+  # code_analysis job runs code analysis scans against the codebase looking for vulnerabilities.
+  code_analysis:
+    name: Run code analysis
+    # This job will not run until build_and_test is complete and passes.
+    needs: build_and_test
+    # Using the latest Ubuntu GitHub-hosted runner.
+    runs-on: ubuntu-latest
+    # Environment variables to use with steps in this job.
+    env:
+      # Use a comma separated list of directories or files to exclude from Snyk test scan.
+      SNYK_EXCLUDES: Tests,Examples,RockLib.Collections.Tests
```

Comment:
How much do you think that the SNYK_EXCLUDES will need to be configured manually? Would it be worth filtering like we do when finding changed projects?

Reply:
Hmm, that's an interesting thought.  Basically filter all the files to get a list of any test or example projects and use that for the SNYK_EXCLUDES value.  I'll probably go back and five that a try.  What I was thinking was that most libraries put the tests in the Tests folder and examples in the Examples folder.  Any project that doesn't do that would need to be added to the exclusion.  I like your idea better though so I'll give that a try.

Reply:
I think you could do something like:
`ls | Where-Object { $_ -Match 'Example.' }`
to get a list of directories and files that begin with "Example."

Reply:
I was able to dynamically generate the exclusion list.  It should catch anything in the Tests or Examples directories as well as any individual test and example projects.